modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Sorting.tsx

```diff
@@ -0,0 +1,344 @@
+
+		if (!field) {
+			alertFailed();
+
+			return null;
+		}
+
+		const response = await fetch(API_URL.FDS_SORTS, {
+			body: JSON.stringify({
+				[OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT_ID]: fdsView.id,
+				fieldName: selectedField,
+				sorting: selectedSorting,
+			}),
+			headers: {
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+			},
+			method: 'POST',
+		});
+
+		if (!response.ok) {
+			alertFailed();
+
+			return null;
+		}
+
+		const responseJSON = await response.json();
+
+		alertSuccess();
+
+		onSave(responseJSON);
+
+		closeModal();
+	};
+
+	return (
+		<div className="fds-view-fields-modal">
+			<ClayModal.Header>
+				{Liferay.Language.get('new-default-sort')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				<ClayForm.Group>
+					<label htmlFor="field">
+						{Liferay.Language.get('field')}
+
+						<RequiredMark />
+					</label>
+
+					<ClaySelectWithOption
+						aria-label={Liferay.Language.get('field')}
+						name="field"
+						onChange={(event) => {
+							setSelectedField(event.target.value);
+						}}
+						options={[
+							{
+								disabled: true,
+								label: Liferay.Language.get('choose-an-option'),
+								selected: true,
+								value: '',
+							},
+							...fields.map((item) => ({
+								label: item.label,
+								value: item.name,
+							})),
+						]}
+						title={Liferay.Language.get('field')}
+						value={selectedField}
+					/>
+				</ClayForm.Group>
+
+				<ClayForm.Group>
+					<label htmlFor="sorting">
+						{Liferay.Language.get('sorting')}
+
+						<RequiredMark />
+					</label>
+
+					<ClaySelectWithOption
+						aria-label={Liferay.Language.get('sorting')}
+						id="sorting"
+						onChange={(event) =>
+							setSelectedSorting(event.target.value)
+						}
+						options={SORTING_OPTIONS}
+					/>
+				</ClayForm.Group>
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton
+							displayType="secondary"
+							onClick={() => closeModal()}
+						>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+
+						<ClayButton
+							disabled={!selectedField}
+							onClick={handleSave}
+						>
+							{Liferay.Language.get('save')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</div>
+	);
+};
+
+const Sorting = ({fdsView, fdsViewsURL}: FDSViewSectionInterface) => {
+	const [fields, setFields] = React.useState<Field[]>([]);
+	const [fdsSorts, setFDSSorts] = useState<Array<FDSSort>>([]);
+	const [newFDSSortsOrder, setNewFDSSortsOrder] = React.useState<string>('');
+
+	useEffect(() => {
+		const getFDSSort = async () => {
+			const response = await fetch(
+				`${API_URL.FDS_VIEWS}/${fdsView.id}?nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT}`
+			);
+
+			const responseJSON = await response.json();
+
+			let fdsSortsOrdered = responseJSON[
+				OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT
+			] as FDSSort[];
```

Comment:
It is usually a good practice to keep variables constant, whenever possible. The more complex the code below becomes, the harder it is to know at a glance what the value of variable represents at certain point in code. I'd keep the response here as an immutable value, `fdsSorts` or `storedFDSSorts`, and build a new array of ordered sorts.